IframeMediaPause/iframe_media_pausing.md

```diff
@@ -121,7 +123,7 @@ oscillator.start(0);
 console.log(audioCtx.state)
 ```
 
-Similarly, for scenario 2, when the iframe is not rendered, the audio context state should change to `'suspended'` and the website can monitor this by listening to the [`statechange`](https://webaudio.github.io/web-audio-api/#eventdef-baseaudiocontext-statechange) event. Then, when the iframe is rendered again, it should wait for a new user interaction, like in the first scenario, to resume playback.
+Similarly, for scenario 2, when the iframe is not rendered, the user agent should run the [`suspend()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspend) steps and set. The audio context state should change to `'suspended'` and the website can monitor this by listening to the [`statechange`](https://webaudio.github.io/web-audio-api/#eventdef-baseaudiocontext-statechange) event. Then, when the iframe is rendered again, it should wait for a new user interaction, like in the first scenario, to resume playback.
```

Comment:
```suggestion
Similarly, for scenario 2, when the iframe is not rendered, the user agent should run the [`suspend()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-suspend) steps. The audio context state should change to `'suspended'` and the website can monitor this by listening to the [`statechange`](https://webaudio.github.io/web-audio-api/#eventdef-baseaudiocontext-statechange) event. Then, when the iframe is rendered again, it should wait for a new user interaction, like in the first scenario, to resume playback.
```

Reply:
In addition, the last sentence doesn't apply when the media element is autoplay enabled. Right?

Reply:
I guess I need to rewrite this. The "should wait for a new user interaction" bit is a recommendation for the callers (not to the specification itself) to avoid calling `resume` without any effects. For example, if the iframe does not wait for a user gesture, it could keep trying to call `resume` while being non-rendered which would fail and keep the `AudioContext` state as `"suspended"`.

Maybe rewrite this to: 
"Then, when the iframe is rendered again, it should call [`resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) and listen for `statechange` events. If the state has not changed to `'running'`, the iframe should call [`resume()`](https://webaudio.github.io/web-audio-api/#dom-audiocontext-resume) in the context of a new user gesture, even though a new user gesture is not strictly necessary to resume callback. We recommend that iframes wait for a new user interaction to avoid repeatedly and unnecessarily calling `resume` while the iframe is still not-rendered."

WDYT?

Reply:
If this is a recommendation for web developers, then a separate paragraph describing what web developers should do as well as some sample code would be appropriate. Note that when you write the spec, such recommendations would go into a non-normative section (ie. a "Note").

I assume you're suggesting that web developers track the non-rendered iframes, listen for `statechange` on them and then call `resume()` after a user gesture. How will web developers know which iframes are rendered vs. not rendered, and how will web developers listen for user gestures?

Reply:
Is the user gesture requirement really just a recommendation for web developers, or is it actually required by the browser to resume playback if there is no permission granted?

Reply:
Unfortunately, we don't know of any ways developers can use to check whether the iframe is rendered or not. Waiting for a new user gesture is the best way we could think of to avoid unsuccessful calls to `resume`.

> Is the user gesture requirement really just a recommendation for web developers, or is it actually required by the browser to resume playback if there is no permission granted?

It is just a recommendation.

Reply:
I have added a code example on what is expected from web pages to do.

Reply:
Discussed offline and decided to remove the "recommendations".